### ISSUE TITLE
Error with workers

### ISSUE THREAD

**Issue by ost89** [MISP/MISP#1088]
After the update to the version 2.4.33 get the next error:
"Unknown    The worker was started with a user other than the apache user. MISP cannot check whether the worker is alive or not."

Additionally, even executed workers are duplicated several times. If click on the "delete/ remove" button, no actions occur.

Please, help

**Comment by iglocska** [comment]
If a worker was started by a different user (for example root from the command line) then the delete / remove buttons indeed won't work, MISP is being denied access to it. 

You can manually stop all workers via the command line and start them again from MISP. The way to do it is as follows:

cd /var/www/MISP/app/Console
./cake CakeResque.CakeResque stop --all 

This should kill all currently running workers, just log into MISP and start the workers from the interface.

Let me know if this doesn't solve it!

**Comment by J3rryT** [comment]
Hi iglocska,

Stopping workers
Stopping 30033 ... Done
Stopping 30015 ... Done
Stopping the Scheduler Worker ... Done
Stopping 30051 ... Done

2nd time:
Stopping workers
   There is no workers to stop ...
- Clearing queues
  There is no queues to clear ...

Still, in the web interface the queues are still present.

Can i  delete/check something in the db (cake_sessions) ?

workers are started by :
su -s /bin/bash %machine account% -c 'scl enable rh-php56 /var/www/MISP/app/Console/worker/start.sh'

We're running on RHEL6 (SeLinux enforcing), so in order to update to MISP 2.4 we where obliged to upgrade php -> 5.6 + apache -> 2.4. 

Btw: I'm a colleague of ost89 :-)

Regards,

Jerry

**Comment by ost89** [comment]
Dear iglocska,

Any updates or news on this?

Thanks!

**Comment by iglocska** [comment]
Sorry completely forgot about this issue, I'll try to look into this together with #1109

**Comment by iglocska** [comment]
Just looked into CentOS/RHEL issues this weekend and I couldn't reproduce the exact problem you have, but I've added a workaround that should allow you to run the workers from the web interface instead of via the command line, so let's try the following:

First of all git pull to get the latest patch.

Simple kill the workers with the command line tool:

```
/var/www/MISP/app/Console/cake CakeResque.CakeResque stop --all
```

The response should be:

```
Stopping workers
Stopping 3675 ... Done
Stopping 3655 ... Done
Stopping the Scheduler Worker ... Done
Stopping 3630 ... Done
```

In the web interface, enable the workaround for SCL (this will inject the rh-php56 path before the execution of the worker start-up scripts, something that has been blocking it from working on CentOS/RHEL). You can do this by Navigating to Administration -> Server settings -> MISP Settings tab and enabling the rh_shell_fix settings.

After this, you can enable all workers by going to Administration -> Server settings -> workers tab and clicking "Restart all workers". Starting and stopping individual workers should work too if you would like to add additional workers.

Let me know if this doesn't solve it for you!

**Comment by J3rryT** [comment]
updated to 2.4.41 ...
and enabled MISP.rh_shell_fix_path.
(content/perms & ownership seems legit)

Stopping the workers using the command line, and restarting using the gui works but the diagnostics page still doesn't detect the cake/worker processes 

the logs look normal and the error log stays untouched ...

[2016-05-10 10:35:35] main.INFO: Exiting... {"type":"shutdown","worker":"s2a900l1:26416"} []
[2016-05-10 10:35:35] main.INFO: Exiting... {"type":"shutdown","worker":"s2a900l1:26397"} []
[2016-05-10 10:35:35] main.INFO: Exiting... {"type":"shutdown","worker":"s2a900l1:26377"} []
[2016-05-10 10:36:00] main.INFO: Starting worker s2a900l1:26768:default {"type":"start","worker":"s2a900l1:26768:default"} []
[2016-05-10 10:36:01] main.INFO: Starting worker s2a900l1:26788:cache {"type":"start","worker":"s2a900l1:26788:cache"} []
[2016-05-10 10:36:01] main.INFO: Starting worker s2a900l1:26806:email {"type":"start","worker":"s2a900l1:26806:email"} []

**Comment by Rafiot** [comment]
Sorry for the late answer, do you still have the issue?

**Comment by rotanid** [comment]
may be related to #1204 - but closing because there was no feedback for 2 months.

**Comment by J3rryT** [comment]
euhm , we still  have this problem. it looks the same in the gui
I also think its related to redis ... ok to close

**Comment by rotanid** [comment]
@J3rryT ah, you're running centos so even if it's the same problem the fix from the other issue wont help... reopening.

**Comment by robertsjw** [comment]
Does anyone have a workaround for this?

**Comment by J3rryT** [comment]
@robertsjw  , @rotanid , I accidentally found a workaround for our setup when troubleshooting some frozen 'workers'.
I've uninstalled redis, uninstalled all php components prior to php56, checked the dependencies, reinstalled redis using pecl.   After this all the issues where resolved.  The workers now correctly report in the workers 'tab' and are perfectly manageable using the web interface.
MISP.rh_shell_fix = True

**Comment by iglocska** [comment]
Great! Could you do a pull request with what was needed to get it resolved? Or alternatively just write some instructions we can use in the FAQ if anyone else runs into the same issue?

**Comment by adulau** [comment]
We will close this issue. If the issue persists, feel free to reopen the ticket.

**Comment by dmolina213** [comment]
I am having the same issue. I am running misp on rhel7 and my version is 2.4.104. when is start the workers i get 
ash-4.2$  /var/www/MISP/app/Console/worker/start.sh 
Stopping workers
Stopping 2264 ... Done
Stopping 2313 ... Done
Stopping 2289 ... Done
Stopping 2240 ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating workers
Starting worker ... Done

Creating the scheduler workers
Starting scheduler worker ..................... Fail
This started showing when I made the change addressed in #1911, changing the setting from false to true.
I could not find MISP.rh_shell_fix_path. in my misp settings
in the workers page I am not seeing the button to start a worker.
